### PR TITLE
[WIP] Handle lambda expressions in Feature Envy

### PR DESCRIPTION
**Pull request by mvz**
#1131.

For now, this just show what I think is the desired behavior. @troessner do you agree that these specs reflect how this should work?

### CODE REVIEW

File: spec/reek/smell_detectors/feature_envy_spec.rb [hunk 1]
```diff
@@ -152,6 +152,36 @@ def alfa
       and reek_of(:FeatureEnvy, name: 'bravo')
   end
 
+  it 'reports calls inside lambda expressions' do
+    src = <<-EOS
+      def alfa(bravo)
+        [
+          -> { bravo.charlie },
+          -> { bravo.delta },
+          -> { bravo.echo },
```
Comment (troessner) [inline spec/reek/smell_detectors/feature_envy_spec.rb]: We shouldn't need this lambda to actually trigger FeatureEnvy, now shouldn't we? The first 2 should be enough following current FeatureEnvy semantics.

File: spec/reek/smell_detectors/feature_envy_spec.rb [hunk 2]
```diff
@@ -152,6 +152,36 @@ def alfa
...
+    expect(src).to reek_of(:FeatureEnvy)
+  end
+
+  it 'does not report different lambda parameters with the same name' do
+    src = <<-EOS
+      def alfa
+        [
+          ->(bravo) { bravo.charlie },
+          ->(bravo) { bravo.delta },
+          ->(bravo) { bravo.echo },
```
Comment (troessner) [inline spec/reek/smell_detectors/feature_envy_spec.rb]: Same here, let's remove this lambda.

**Comment by troessner** [comment]
Couple of smaller comments, overall I agree that this is the way it should behave.

File: spec/reek/smell_detectors/feature_envy_spec.rb (hunk 1 above)
Comment (mvz) [reply spec/reek/smell_detectors/feature_envy_spec.rb]: It seemed two was not enough, but I'll check again.

**Comment by troessner** [comment]
@mvz this PR seems stale, wdyt about closing it for now?

**Comment by mvz** [comment]
@troessner yes, agreed.

**Pull request closed without merging**